src/components/GitStash.tsx

```diff
@@ -0,0 +1,386 @@
+   */
+  files: string[];
+  /**
+   * Index within the stash
+   */
+  index: number;
+  /**
+   * HTML element height
+   */
+  height: number;
+  /**
+   * Optional select all element
+   */
+  selectAllButton?: React.ReactElement;
+
+  /**
+   * Branch corresponding to the stash
+   */
+  branch: string;
+
+  /**
+   * message corresponding to the stash
+   */
+  message: string;
+
+  /**
+   * Wrap mouse event handler to stop event propagation
+   * @param fn Mouse event handler
+   * @returns Mouse event handler that stops event from propagating
+   */
+  stopPropagationWrapper: (
+    fn: React.EventHandler<React.MouseEvent>
+  ) => React.EventHandler<React.MouseEvent>;
+}
+
+/**
+ * Dropdown for each entry in the stash
+ */
+const GitStashEntry: React.FunctionComponent<IGitStashEntryProps> = (
+  props: IGitStashEntryProps
+) => {
+  const [showStashFiles, setShowStashFiles] = React.useState(false);
+
+  const nFiles = props?.files?.length;
+
+  // const gitStashPop = async (index: number): Promise<void> => {
+  //   try {
+  //     await props.model.stash_pop(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  // const gitStashDrop = async (index: number): Promise<void> => {
+  //   console.log('Call Git Stash Drop', index);
+  //   try {
+  //     await props.model.stash_drop(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  // const gitStashApply = async (index: number): Promise<void> => {
+  //   try {
+  //     await props.model.stash_apply(index);
+  //   } catch (err) {
+  //     console.error(err);
+  //   }
+  // };
+
+  const getFilePath = (file: string) => {
+    // Root directory
+    if (!file.includes('/')) {
+      return {
+        name: file,
+        path: ''
+      };
+    } else {
+      // One folder
+      const fileSplit = file.split('/');
+      if (fileSplit.length == 2) {
+        return {
+          path: fileSplit[0],
+          name: fileSplit[1]
+        };
+        // More than one nested folder
+      } else {
+        return {
+          name: fileSplit.pop(),
+          path: fileSplit.join('/')
+        };
+      }
+    }
+  };
+
+  return (
+    <div className={sectionFileContainerStyle}>
+      <div
+        className={sectionAreaStyle}
+        onClick={() => {
+          if (props.collapsible && props?.files.length > 0) {
+            setShowStashFiles(!showStashFiles);
+          }
+        }}
+      >
+        {props.collapsible && (
+          <button className={changeStageButtonStyle}>
+            {showStashFiles && props?.files.length > 0 ? (
+              <caretDownIcon.react />
+            ) : (
+              <caretRightIcon.react />
+            )}
+          </button>
+        )}
+        <span
+          className={sectionHeaderLabelStyle}
+          style={{ display: 'flex', justifyContent: 'space-between' }}
+        >
+          <p>
+            {props.message} (on {props.branch})
```

Comment:
You need to pass the translator object through the react tree and change this to:

```suggestion
            {props.trans.__('%1 (on %2)', props.message, props.branch)}
```

Reply:
Do we really want to translate the props message? If the user makes their own message, shouldn't we assume it's in their native language?